src/odemis/driver/autoscript_client.py

```diff
@@ -0,0 +1,2122 @@
+        :param name (str): name of the Detector
+        :param role (str): role of the Detector
+        :param parent (SEM): parent of the Detector
+        :param channel (str): name of the acquistion channel (electron, ion)
+        :param kwargs: additional keyword arguments
+        """
+        # The acquisition is based on a FSM that roughly looks like this:
+        # Event\State |    Stopped    |   Acquiring    | Receiving data |
+        #    START    | Ready for acq |        .       |       .        |
+        #    DATA     |       .       | Receiving data |       .        |
+        #    STOP     |       .       |     Stopped    |    Stopped     |
+        #    TERM     |     Final     |      Final     |     Final      |
+        self.parent: SEM  # for type hinting
+        model.Detector.__init__(self, name, role, parent=parent, **kwargs)
+        self._shape = (256,)  # Depth of the image
+        self.data = SEMDataFlow(self)
+        self.channel = channel
+
+        if self.channel == "electron":
+            self._scanner = self.parent._scanner
+        elif self.channel == "ion":
+            self._scanner = self.parent._fib_scanner
+        else:
+            raise ValueError("Invalid channel name. No scanner available for this channel.")
+
+        brightness_info = self.parent.brightness_info(self.channel)
+        self.brightness = model.FloatContinuous(
+            self.parent.get_brightness(self.channel),
+            brightness_info["range"],
+            unit=brightness_info["unit"],
+            setter=self._setBrightness)
+
+        contrast_info = self.parent.contrast_info(self.channel)
+        self.contrast = model.FloatContinuous(
+            self.parent.get_contrast(self.channel),
+            contrast_info["range"],
+            unit=contrast_info["unit"],
+            setter=self._setContrast)
+
+        detector_type = self.parent.get_detector_type(self.channel)
+        self.detector_type = model.StringEnumerated(
+            detector_type,
+            choices=set(self.parent.detector_type_info(self.channel)["choices"]),
+            setter=self._setDetectorType)
+
+        detector_mode = self.parent.get_detector_mode(self.channel)
+        self.detector_mode = model.StringEnumerated(
+            detector_mode,
+            choices=set(self.parent.detector_mode_info(self.channel)["choices"]),
+            setter=self._setDetectorMode)
+
+        self._genmsg = queue.Queue()  # GEN_*
+        self._generator = None
+
+        # Refresh regularly the values, from the hardware, starting from now
+        self._updateSettings()
+        # self._va_poll = util.RepeatingTimer(5, self._updateSettings, "Settings polling detector")
+        # self._va_poll.start()
+        # Note: using the repeated polling causes the view in xtUI to be changed every five seconds,
+        # this makes it pretty annoying to do anything on that side, and error prone.
+        # disabling this until a better solution is found
```

Comment:
I think if we do that it would be confusing, because user might 'select' the SEM view in odemis or just leave it on that view, then start using the xtui and it would automatically switch back which would be confusing/annoying. It would also happen if user leaves odemis running which would be even more confusing.

In practice, this only updates detector brightness, contrast, mode and type. mode and type are usually just set once, and not changed. And brightness / contrast change when the user changes them or runs auto-brightness-contrast, so we could just update them directly there? We can see how required the auto-updates are after that

Reply:
sounds good